src/main/java/uk/gov/ons/ctp/integration/ratelimiter/client/RateLimiterClient.java

```diff
@@ -153,32 +176,37 @@ public RateLimitResponse checkRateLimit(
     return response;
   }
 
+  // Throws CTPException is the argument is null
   private void verifyArgumentSupplied(String argName, Object argValue) throws CTPException {
     if (argValue == null) {
       throw new CTPException(Fault.SYSTEM_ERROR, "Argument '" + argName + "' cannot be null");
     }
   }
 
+  // Throws CTPException if an argument is supplied but it is blank
   private void verifyArgumentNotEmpty(String argName, String argValue) throws CTPException {
     if (argValue != null && argValue.isBlank()) {
       throw new CTPException(
           Fault.SYSTEM_ERROR, "Argument '" + argName + "' cannot be blank (" + argValue + ")");
     }
   }
 
+  // This is a key method that bunches together the various arguments that the limiter will be using
+  // to decide if the request has breached any limits.
   private RateLimitRequest createRateLimitRequest(
```

Comment:
This is a problem observed in the original code, but worth fixing here.
As discussed in our call, the exception "catch" clause here is a bit misleading because `response` is reused purely for building a logging string. 
I recommend pulling out the code used to create this line ` log.info(failureDescription.toString());` into a private method. 
Also as you mentioned Pete, be more clear about where the ResponseStatusException is thrown , since at present throwing it after the CTPException makes it less clear.

Reply:
Agree. 
Extracted the what-went-wrong description to its own method.